landoapi/landing_worker.py

```diff
@@ -435,9 +456,20 @@ def run_job(
                     self.notify_user_of_landing_failure(job)
                     return False
 
-            commit_id = hgrepo.run_hg(["log", "-r", ".", "-T", "{node}"]).decode(
-                "utf-8"
-            )
+            # Gather changesets hashes and commit titles for each node in the stack in
+            # case we need to further analyse. Use the null character as the separator.
+            # This variable is a list of (node, changeset title) pairs.
+            stack_ids: list[tuple[str, str]] = [
+                tuple(line.split("\0", 1))
+                for line in hgrepo.run_hg(
+                    ["log", "-r", "stack()", "-T", "{node}\\0{desc|firstline}\n"]
+                )
+                .decode("utf-8")
+                .splitlines()
+            ]
```

Comment:
I would suggest moving this to where it is used below (and perhaps not including `{node}` since it does not appear to be used). And, keeping the original `commit_id` definition here. Sure we would be running two separate commands, but this command is less relevant for the general code flow and the node IDs are mostly not needed, with the exception of the first one.

Reply:
`node` is used in `commit_id` right below. I figured getting all the information about the stack from `hg log` would be the most efficient but running the command twice works too.